src/snowflake/cli/_plugins/snowpark/snowpark_project_paths.py

```diff
@@ -74,19 +78,29 @@ def snowflake_requirements(self) -> SecurePath:
     def requirements(self) -> SecurePath:
         return SecurePath(self.path_relative_to_root(Path("requirements.txt")))
 
+    @property
+    def deploy_root(self) -> Path:
+        return self.project_root / "output" / "deploy" / "snowpark"
+
```

Comment:
This path is build in multiple places in repeated manner.
How about creating a common tool for a deploy root path

```python
from pathlib import Path


def deploy_root(root: Path, app_type: str | None = None) -> Path:
    if app_type:
        return root / "output" / "deploy" / app_type
    return root / "output" / "deploy"
```
---

```python
@property
def deploy_root(self) -> Path:
   return deploy_root(self.project_root, "snowpark")
```
---

```python
@property
def deploy_root(self) -> Path:
   return deploy_root(self.project_root, "streamlit")
```
---
```python
@property
def deploy_root(self) -> Path:
   return deploy_root(self.project_root)
```
---

Reply:
Changed